include/nvqpp_config.h.in

```diff
@@ -7,13 +7,17 @@
  ******************************************************************************/
 
 #pragma once
-#define PLATFORM_SHARED_LIBRARY_SUFFIX "${CMAKE_SHARED_LIBRARY_SUFFIX}"
-#define LLVM_INCLUDE_DIR "${LLVM_INCLUDE_DIRS}"
-#define LLVM_ROOT "${LLVM_BINARY_DIR}"
-#define LLVM_LIBCXX_INCLUDE_DIR "${LLVM_BINARY_DIR}/include/c++/v1"
-#define CUDAQ_LLVM_VERSION "${CUDAQ_LLVM_VERSION}"
+#define PLATFORM_SHARED_LIBRARY_SUFFIX "@CMAKE_SHARED_LIBRARY_SUFFIX@"
+#define LLVM_INCLUDE_DIR "@LLVM_INCLUDE_DIRS@"
+#define LLVM_ROOT "@LLVM_BINARY_DIR@"
+#ifndef LLVM_CONFIG_H
+#include "llvm/Config/llvm-config.h"
+#endif
+#ifndef LLVM_TARGET_TRIPLE
+#define LLVM_TARGET_TRIPLE LLVM_DEFAULT_TARGET_TRIPLE
```

Comment:
Hm, trying to think of the scenarios here, and which ones we want to support: 
- Case 1: the header file is used by code compiled with CUDA-Q. In this case all is good since the LLVM version is the one we indeed rely on. 
- Case 2: the header file is used by code compiled with an external compiler. In this case, LLVM_TARGET_TRIPLE is set by the llvm-config.h, which we have to be able to include (if it is not included here, it is still included by other parts of the code base). LLVM_TARGET_TRIPLE, or any other LLVM variables, necessarily needs to be defined by the LLVM version that CUDA-Q relies on, I believe. If this were to interfere with the LLVM version by other parts of the build I think we have a problem in general - you then have a project that utilizes multiple version of LLVM in the same compilation unit. I don't think this is going to work, regardless of what variable names we use.